src/Configuration/Programmer/Programmer_helper.cpp

```diff
@@ -0,0 +1,453 @@
+        cable.productId == std::get<1>(supportedCable)) {
+      cableSupported = true;
+      break;
+    }
+  }
+  if (!cableSupported) {
+    return ProgrammerErrorCode::CableNotSupported;
+  }
+  return ProgrammerErrorCode::NoError;
+}
+
+/*
+This function expecting the string from openOCD scan_chain output
+with example below
+
+testString =
+"   TapName            Enabled IdCode     Expected   IrLen IrCap IrMask\n"
+"-- ------------------ ------- ---------- ---------- ----- ----- ------\n"
+"0  omap5912.dsp          Y    0x03df1d81 0x03df1d81 38    0x01  0x03\n"
+"1  omap5912.arm          Y    0x0692602f 0x0692602f 4     0x01  0x0f\n"
+"2  omap5912.unknown      Y    0x00000000 0x00000000 8     0x01  0x03\n"
+"3 auto0.tap              Y     0x20000913 0x00000000     5 0x01  0x03
+^\d+\s+\w+\.\w+\s+\w+\s+0x[\da-fA-F]+\s+0x[\da-fA-F]+\s+\d+\s+0x[\da-fA-F]+\s+0x[\da-fA-F]+$
+*/
+std::vector<TapInfo> extractTapInfoList(const std::string& tapInfoString) {
+  std::vector<TapInfo> tapInfoList;
+  std::istringstream iss(tapInfoString);
+  std::string line;
+  std::string token;
+  size_t pos = -1;
+  std::vector<std::string> matches;
+  const std::string pattern(
+      R"(\s*(\d+)\s+(\S+)\s+(\S+)\s+(\S+)\s+(\S+)\s+(\d+)\s+(\S+)\s+(\S+)\s*)");
+  while (std::getline(iss, line)) {
+    matches = findStringPattern(line, pattern);
+    if (matches.size() == 0) {
+      continue;
+    }
+    // Replace all the spaces with a single space
+    while ((pos = line.find("  ")) != std::string::npos) {
+      line.replace(pos, 2, " ");
+    }
+    std::istringstream lineStream(line);
+    std::vector<std::string> tokens;
+    while (lineStream >> token) {
+      tokens.push_back(token);
+    }
+    if (tokens.size() == 8) {
+      TapInfo tapInfo;
+      tapInfo.index = CFG_convert_string_to_u64(tokens[0]);
+      tapInfo.tapName = tokens[1];
+      tapInfo.enabled = tokens[2] == "Y" ? true : false;
+      tapInfo.idCode =
+          static_cast<uint32_t>(CFG_convert_string_to_u64(tokens[3], 0, 0));
+      tapInfo.expected =
+          static_cast<uint32_t>(CFG_convert_string_to_u64(tokens[4], 0, 0));
+      tapInfo.irLen =
+          static_cast<uint32_t>(CFG_convert_string_to_u64(tokens[5]));
+      tapInfo.irCap =
+          static_cast<uint32_t>(CFG_convert_string_to_u64(tokens[6], 0, 0));
+      tapInfo.irMask =
+          static_cast<uint32_t>(CFG_convert_string_to_u64(tokens[7], 0, 0));
+      tapInfoList.push_back(tapInfo);
+    }
+  }
+
+  return tapInfoList;
+}
+/*
+This function expecting the string from openOCD gemini list output
+with example below
+
+         Device               ID           IRLen      Flash Size
+-------- -------------------- ------------ ---------- ----------------
+Found  0 Gemini               0x20000913   5          16384
+*/
+std::vector<Device> extractDeviceList(const std::string& devicesString) {
+  std::vector<Device> devices;
+
+  std::istringstream iss(devicesString);
+  std::string line;
+  size_t pos = -1;
```

Comment:
size_t is unsigned type. Looks suspicious

Reply:
Fixed. Change to std::string:npos